doc/pmon/sonic_thermal_control_test_plan.md

```diff
@@ -0,0 +1,250 @@
+    ]
+}
+```
+
+#### Invalid format policy file
+
+In invalid_format_policy.json, the file content is not valid JSON at all. A file with content "invalid" should be good for test purpose.
+
+#### Invalid value policy file
+
+In invalid_value_policy.json, the file content contains minus value of target speed. We couldn't cover all invalid value here because there are too many possibilities. The major purpose of this configuration file is to verify thermal control daemon would not crash while loading such an invalid configuration files. For other negative test cases, they are already covered by unit test. The invalid_value_policy.json content is like:
+
+```json
+{
+    "info_types": [
+        {
+            "type": "fan_info"
+        },
+        {
+            "type": "psu_info"
+        }
+    ],
+    "policies": [
+        {
+            "name": "any PSU absence",
+            "conditions": [
+                {
+                    "type": "fan.any.absence"
+                }
+            ],
+            "actions": [
+                {
+                    "type": "fan.all.set_speed",
+                    "speed": "-1"
+                }
+            ]
+        }
+    ]
+}
+```
+
+## Test Cases
+
+### Show FAN Status Test
+
+Show FAN status test verifies that all FAN related information can be shown correctly via `show platform fanstatus`.
+
+#### Procedure
+
+1. Testbed setup.
+2. Unlink FAN related sysfs and fill mock data for "presence", "speed", "status", "target_speed", "led status".
+3. Issue command `show platform fanstatus`.
+4. Record the command output.
+5. Verify that command output matches the mock data.
+6. Restore FAN related soft links in sysfs.
+
+### Show Thermal Status Test
+
+Show thermal status test verifies that all thermal related information can be shown correctly via `show platform temperature`.
+
+#### Procedure
+
+1. Testbed setup.
+2. Unlink thermal related sysfs and fill mock data for "temperature", "high_threshold", "high_critical_threshold".
+3. Issue command `show platform temperature`.
+4. Record the command output.
+5. Verify that command output matches the mock data.
+6. Restore FAN related soft links in sysfs.
+
+### FAN Test
+
+FAN test verifies that proper action should be taken for conditions including: FAN absence, FAN over speed, FAN under speed.
+
+#### Procedure
+
+1. Testbed setup.
+2. Copy valid_policy.json to pmon docker and backup the original one.
+3. Restart pmon service by command `systemctl restart pmon` to trigger thermal control daemon reload policy configuration file.
+4. Unlink FAN related sysfs and make mock data: first FAN absence.
+5. Wait for at least 65 seconds. Verify target speed of all FANs are set to 100% according to valid_policy.json. Verify there is a warning log for FAN absence.
+6. Make mock data: first FAN presence.
+7. Wait for at least 65 seconds. Verify target speed of all FANs are set to 60% according to valid_policy.json. Verify there is a notice log for FAN presence.
```

Comment:
i think 60% is not a good number.
when thermal control is disabled this is the default system value (at least in mellanox switches) thus if this comes from the config file, i suggest to use other value.